File: src/Backend/LibrarySystem.Domain/Entities/CourseReserves/Course.cs
using System.Collections.Generic;

namespace LibrarySystem.Domain.Entities.CourseReserves
{
    public class Course
    {
        public int Id { get; set; }
        public string Code { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        
        public int InstructorId { get; set; }
        public Instructor Instructor { get; set; } = null!;

        public ICollection<CourseReserveList> ReserveLists { get; set; } = new List<CourseReserveList>();
    }
}

File: slib-frontend-monorepo/packages/types/src/course.ts
// types for course reserves

export interface Instructor {
    id: number;
    name: string;
    email: string;
    department: string;
}

export interface Course {
    id: number;
    code: string;
    name: string;
    instructorId: number;
    instructor?: Instructor;
    reserveLists?: CourseReserveList[];
}

export interface CourseReserveList {
    id: number;
    courseId: number;
    term: string;
    status: string;
    activeFrom: string;
    activeTo: string;
    items?: CourseReserveItem[];
}

export interface CourseReserveItem {
    id: number;
    courseReserveListId: number;
    physicalItemId: number;
    reservePolicy: string;
    physicalItem?: any; // To avoid circular imports for now, we can use any or define PhysicalItem here
}

export interface AddReserveItemRequest {
    courseReserveListId: number;
    physicalItemId: number;
    reservePolicy: string;
}
